pkg/scaler/prober.go

```diff
@@ -260,6 +270,31 @@ func (p *prober) getProbeResultLabels(pr *probeResult) prometheus.Labels {
 // 6. If the external probe is HEALTHY then the dependants are scaled up.
 // 7. If the external probe is UNHEALTHY then the dependants are scaled down.
 func (p *prober) probe(ctx context.Context) error {
+	// Check state of the cluster. The name of cluster is same as shoot's namespace
+	clusterName := p.namespace
+	if cluster, err := p.clusterLister.Get(clusterName); apierrors.IsNotFound(err) {
+		klog.Errorf("Could not find cluster: %s, Stopping probe", clusterName)
+		return err
+	} else if err != nil {
+		klog.Errorf("Error fetching cluster: %s. Err: %s", clusterName, err)
+		return nil
+	} else {
+		decoder, err := extensionscontroller.NewGardenDecoder()
+		if err != nil {
+			return nil
+		}
+
+		shoot, err := extensionscontroller.ShootFromCluster(decoder, cluster)
+		if err != nil {
+			return nil
+		}
+		if shoot.Spec.Hibernation == nil || shoot.Spec.Hibernation.Enabled == nil || (*shoot.Spec.Hibernation.Enabled == false && shoot.Status.IsHibernated == false) {
```

Comment:
Shouldn't this be the following?

```suggestion
		if (shoot.Spec.Hibernation == nil || shoot.Spec.Hibernation.Enabled == nil || *shoot.Spec.Hibernation.Enabled == false) && shoot.Status.IsHibernated == false {
```